### DISCUSSION TITLE
Global loading state?

### DISCUSSION THREAD

**Discussion by mitchumowen** [livewire/livewire#2045] [1. Help]
I'm aware of the component-level <wire:loading> functionality, but my understanding is that this only applies to the loading of a given component.  Has anyone successfully implemented something like <wire:loading-global> that shows when any Livewire component on the page is waiting for a response?

**Comment by artworkad** [comment]
@mitchumowen please refer to https://github.com/livewire/livewire/discussions/2355

**Comment by calebporzio** [comment]
Curious what the use case is?

**Reply by artworkad** [reply]
@calebporzio use case is to give the user some feedback "this is taking longer than expected". For me specifically it's a reusable livewire modal which can load any other livewire component. Loading of child components inside the modal is fast, but when it takes more than 250 ms I would like to show some loading indicator. On the other side, I would see the necessity of such a loading indicator as a sign of a flaw in the application design itself and consider working on this problem instead of going just for the loading indicator.
